clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/KubernetesV2ProviderSynchronizable.java

```diff
@@ -117,17 +118,38 @@ public void synchronize() {
                 // account exists but has changed
                 changedAccounts.add(managedAccount.getName());
                 newAndChangedAccounts.add(managedAccount.getName());
+              } else {
+                // Current credentials may contain memoized namespaces, we should keep if the
+                // definition has not changed
+                return null;
               }
-
-              accountCredentialsRepository.save(managedAccount.getName(), credentials);
-            });
+              return credentials;
+            })
+        .filter(Objects::nonNull)
+        .forEach(a -> saveToCredentialsRepository(a));
 
     ProviderUtils.unscheduleAndDeregisterAgents(deletedAccounts, catsModule);
     ProviderUtils.unscheduleAndDeregisterAgents(changedAccounts, catsModule);
 
     return newAndChangedAccounts;
   }
 
+  /**
+   * Validate and save credentials to repository
+   *
+   * @param account
+   */
+  protected void saveToCredentialsRepository(KubernetesNamedAccountCredentials account) {
```

Comment:
I think this can be private?

Reply:
I was hoping to keep it `protected` to have options that performs different levels of validations but we can revisit if needed. Updated to `private`